src/cbor.rs

```diff
@@ -0,0 +1,227 @@
+use coset::{cbor, CoseError, EndOfFile};
+use serde::{de, Deserialize, Serialize};
+use std::borrow::{Borrow, BorrowMut};
+use std::io::Cursor;
+use std::ops::{Deref, DerefMut};
+use thiserror::Error;
+
+/// Wraps [ciborium::Value] and implements [PartialEq], [Eq], [PartialOrd] and [Ord],
+/// so it can be used in maps and sets.
+///
+/// Also, useful in future if we want to change the CBOR library.
+#[derive(Debug, Clone)]
+pub struct Value(pub ciborium::Value);
```

Comment:
Resolved this by checking for `non-finite` values and return `Err` from constructor. Have another `unsafe` constructor which doesn't do the check.

Reply:
My main question was: do we need this type?
The only instance where you seem to be constructing it is in `NfcOptions: TryFrom<Value>`, but it doesn't seem necessary given that `NfcOptions` doesn't contain any `Value`.

Reply:
It's used is some places in `collections` where we need `Ord` and `Eq`.

```
src/definitions/device_key/mod.rs:66
pub key_info: Option<BTreeMap<i128, CborValue>>,

src/issuance/mdoc.rs:24
pub type Namespaces = BTreeMap<String, BTreeMap<String, CborValue>>;

src/definitions/device_signed.rs
pub type DeviceSignedItems = NonEmptyMap<String, CborValue>;

src/issuance/mdoc.rs
pub type Namespaces = BTreeMap<String, BTreeMap<String, CborValue>>;

```
And in other places as well where we deserialize from `ciborium::Value` we build a `Map<CborValue, CborValue>` from the `Vec<(ciborium::Valu, ciborium::Value)>`.

So I think we still need it, but if you see it otherwise let me know how to resolve the above cases.

Reply:
In this later example, I think building a map is unnecessary.
You could visit the `map` in one pass to extract the values you need. For instance for `device_engagement_version`:
```rust
let device_engagement_version_key = ciborium::Value::Integer(0.into());
let mut device_engagement_version_value = None;

for (key, value) in map { // visit the map
  if key == device_engagement_version_key { // found the key we need
    device_engagement_version_value = Some(value)
  }
}
```

Reply:
I'm using `i128` as key now.